storage/connectors/bigquery/src/main/java/feast/storage/connectors/bigquery/statistics/Dataset.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.storage.connectors.bigquery.statistics;
+
+import com.google.cloud.bigquery.TableId;
+import com.google.protobuf.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+/**
+ * Dataset represents a subset of data within a table to retrieve statistics over. Data can be
+ * subset by either ingestionId or date.
+ */
+public class Dataset {
```

Comment:
I'm a bit iffy about dataset being used as the term for what is more comparable to your `Info` value types. That being said I think this is ok for now but this would be tech debt that we will clear later once we have proper datasets as a concept.

Reply:
I don't really like it either. Any suggestions? It was initially DataSubset but the two are pretty analogous to each other

Reply:
Dunno, maybe just taint it with some prefix. We can always pull something out of our bag of generic suffixes.

`StatsDatasetInfo`?

Sounds horrible, but at least it doesn't name squat `Dataset` :)